Filter tarball files with globs; filter more files

V-FEXrt opened the pull request:
Part two of splitting up https://github.com/sifive/wake/pull/1364

This change 
 - removes more files from the  tarball that have snuck in over time
 - uses globs  for exclusion instead of regexes
 - prints the excluded files to `info`

JakeSiFive commented on share/wake/lib/core/regexp.wake:
```diff
@@ -81,6 +81,28 @@ export def regExpToString (regExp: RegExp): String =
 export def matches (testRegExp: RegExp) (str: String): Boolean =
     (\_ \_ prim "match") testRegExp str
 
+# Test if a String matches all provided regular expressions
+# matchesAll (`a*`, `*b`) "ab" = True
+# matchesAll (`a*`, `*b`) "a" = False
+export def matchesAll (testRegExps: List RegExp) (str: String): Boolean = match testRegExps
+    Nil = True
+    head, tail =
+        if (matches head str) then
+            matchesAll tail str
+        else
+            False
```
This is too generic to be in regex, use forall/exist in list.wake

V-FEXrt replied:
ah ye. makes sense. fixed!

JakeSiFive commented on share/wake/lib/core/regexp.wake:
```diff
@@ -81,6 +81,28 @@ export def regExpToString (regExp: RegExp): String =
+# Test if a String matches any provided regular expressions
+# matchesAny (`a*`, `*b`) "ac" = True
+# matchesAny (`a*`, `*b`) "c" = False
+export def matchesAny (testRegExps: List RegExp) (str: String) = match testRegExps
```
ditto

V-FEXrt commented:
@JakeSiFive I'm going to go ahead and land this since I've addressed your comments and this PR is fairly small/uncontroversial now. If you have any changes you want when you get  back lmk and I'll make them asap
